### PR TITLE
Add SmartyPants flag

### LINKED ISSUE #5752: Add smartypants back as a flag / option
option name?

- smartypants: true?
- typographer: true?
- something else

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes

Resolves #5752 

SmartyPants was removed from Astro's defaults per #5684. This adds back our default with a `markdown.smartypants` option to disable.

## Testing

Add SmartyPants tests for Markdown and MDX

## Docs

- Update configuration API
- Update docs reference https://github.com/withastro/docs/pull/2301

### CODE REVIEW

File: .changeset/angry-pots-boil.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+'astro': patch
+'@astrojs/mdx': patch
+'@astrojs/markdown-remark': patch
+---
+
+Add SmartyPants back to Astro's default Markdown and MDX plugins. This can be disabled using the `markdown.smartypants` config option:
```
Comment (natemoo-re) [inline .changeset/angry-pots-boil.md:7]: I would reword this changelog from the perspective of migrating from Astro 1.0 => Astro 2.0!

**Review by natemoo-re — COMMENTED** [review]
Code looks good!

**Comment by matthewp** [comment]
Were other naming choices discussed (I remember `typographer` being one)? Can you explain that decision?

**Comment by delucis** [comment]
> Were other naming choices discussed (I remember typographer being one)? Can you explain that decision?

Seems like it would be odd to name it something different from the plugin we’re using? Unless we were planning to roll-our-own custom bundle of typography related plugins.

**Comment by matthewp** [comment]
@bholmesdev can you mark this as closing https://github.com/withastro/astro/issues/5752? thanks.

File: packages/astro/src/@types/astro.ts [hunk 2]
```diff
@@ -785,6 +785,23 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		gfm?: boolean;
+		/**
+		 * @docs
```
Comment (sarah11918) [inline packages/astro/src/@types/astro.ts:789]: Docs approval here!

**Comment by bholmesdev** [comment]
@matthewp Discussed this with @natemoo-re! In short, Goldmark is the only library I could find using "typographer" as their option name. "smartypants" has documentation across Python, Drupal, and JS Markdown libraries from an initial search. Sticking with `smartypants` makes sense for that reason.

File: .changeset/angry-pots-boil.md (hunk 1 above)
Comment (bholmesdev) [reply .changeset/angry-pots-boil.md]: Done!

**Review by matthewp — APPROVED** [review]

**Review by natemoo-re — APPROVED** [review]

**Comment by bholmesdev** [comment]
@matthewp @natemoo-re Double checking what the version for this change should be? I chose "patch" instead of "minor" since it's a 2.0 beta feature.
